build.gradle

```
@@ -127,66 +127,90 @@ subprojects {
         }
     }
 
-    uploadArchives {
-        repositories {
-            mavenDeployer {
-                if (sonatype_publish) {
-                    beforeDeployment { 
-                        MavenDeployment deployment -> signing.signPom(deployment) 
-                    }
-
-                    repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2") {
-                        authentication(userName: ossrhUsername, password: ossrhPassword)
-                    }
-
-                    snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots") {
-                        authentication(userName: ossrhUsername, password: ossrhPassword)
-                    }
+    publishing {
+        publications {
+            mavenJava(MavenPublication) {
+                from components.java
 
-                    pom.project {
-                        name 'Exactpro QuickFixJ'
+                if (sonatype_publish) {
+                    pom {
+                        name = 'Exactpro QuickFixJ'
                         packaging 'jar'
                         // optionally artifactId can be defined here 
                         description 'QuickFixJ is one of such libraries we depend on and which was modified by Exactpro.'
-                        url 'https://github.com/Exactpro/quickfixj'
+                        url = vcsUrl
 
                         scm {
-                            connection 'scm:git:https://github.com/Exactpro/quickfixj'
-                            developerConnection 'scm:git:https://github.com/Exactpro/quickfixj'
-                            url 'https://github.com/Exactpro/quickfixj'
+                            connection = 'scm:git:https://github.com/Exactpro/quickfixj'
+                            developerConnection = 'scm:git:https://github.com/Exactpro/quickfixj'
+                            url = vcsUrl
                         }
 
                         licenses {
                             license {
-                                name 'The Apache License, Version 2.0'
-                                url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                                name = 'The Apache License, Version 2.0'
+                                url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
                             }
                         }
 
                         developers {
                             developer {
-                                id 'Nikita-Smirnov-Exactpro'
-                                name 'Nikita Smirnov'
-                                email 'nikita.smirnov@exactprosystems.com'
+                                id = 'Nikita-Smirnov-Exactpro'
+                                name = 'Nikita Smirnov'
+                                email = 'nikita.smirnov@exactprosystems.com'
```

```suggestion
                                id = 'sf_devs'
                                name = 'sf_devs'
                                email = 'sf_devs@exactpro.com'
```

                                id.set("developer")
                                name.set("developer")
                                email.set("developer@exactpro.com")